TrenchBroomTest/raylib_viewer.odin

```
@@ -0,0 +1,471 @@
+		initial_pos = rl.Vector3 {
+			spawn.position.x * MAP_SCALE,
+			spawn.position.z * MAP_SCALE + 1,
+			-spawn.position.y * MAP_SCALE,
+		}
+		fmt.printf("Starting at spawn point: %v (scaled: %v)\n", spawn.position, initial_pos)
+	} else {
+		center := (loaded_map.map_bounds.min + loaded_map.map_bounds.max) * 0.5
+		initial_pos = rl.Vector3 {
+			center.x * MAP_SCALE,
+			center.z * MAP_SCALE + 5,
+			-center.y * MAP_SCALE,
+		}
+		fmt.printf("Starting at map center: %v (scaled: %v)\n", center, initial_pos)
+	}
+
+	camera = rl.Camera3D {
+		position   = initial_pos,
+		target     = rl.Vector3{initial_pos.x, initial_pos.y - 1, initial_pos.z - 1},
+		up         = rl.Vector3{0, 1, 0},
+		fovy       = 75.0,
+		projection = .PERSPECTIVE,
+	}
+
+	camera_yaw = 0.0
+	camera_pitch = 0.0
+}
+
+load_textures :: proc() {
+	loaded_textures = make(map[string]rl.Texture2D)
+
+	texture_dir := "textures"
+	dir_handle, dir_err := os.open(texture_dir)
+	if dir_err != os.ERROR_NONE {
+		fmt.printf("Failed to open textures directory: %v\n", dir_err)
+		return
+	}
+	defer os.close(dir_handle)
+
+	file_infos, read_err := os.read_dir(dir_handle, -1)
+	if read_err != os.ERROR_NONE {
+		fmt.printf("Failed to read textures directory: %v\n", read_err)
+		return
+	}
+	defer delete(file_infos)
+
+	for file_info in file_infos {
+		if strings.has_suffix(file_info.name, ".png") {
+			texture_name := strings.trim_suffix(file_info.name, ".png")
+			texture_path := fmt.tprintf("%s/%s", texture_dir, file_info.name)
+
+			texture_path_cstr := strings.clone_to_cstring(texture_path, context.temp_allocator)
+
+			texture := rl.LoadTexture(texture_path_cstr)
+			if texture.id != 0 {
+				loaded_textures[strings.clone(texture_name)] = texture
+				fmt.printf("Loaded texture: %s\n", texture_name)
+			} else {
+				fmt.printf("Failed to load texture: %s\n", texture_path)
+			}
+		}
+	}
+
+	fmt.printf("Loaded %d textures\n", len(loaded_textures))
+}
+
+convert_quake_meshes_to_raylib :: proc() {
```

This should return an error enum